pkg/render/goldmane/component.go

```diff
@@ -0,0 +1,320 @@
+	}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        GoldmaneDeploymentName,
+			Namespace:   GoldmaneNamespace,
+			Annotations: annotations,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.ToPtr(int32(1)),
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: GoldmaneDeploymentName,
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:       c.cfg.Installation.ControlPlaneNodeSelector,
+					ServiceAccountName: GoldmaneServiceAccountName,
+					Tolerations:        tolerations,
+					ImagePullSecrets:   secret.GetReferenceList(c.cfg.PullSecrets),
+					Containers:         ctrs,
+					Volumes:            volumes,
+				},
+			},
+		},
+	}
+}
+
+func (c *Component) roleBinding() *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GoldmaneRoleName,
+			Namespace: GoldmaneNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     GoldmaneRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      GoldmaneRoleName,
+				Namespace: GoldmaneNamespace,
+			},
+		},
+	}
+}
+
+func (c *Component) role() *rbacv1.Role {
+	policyRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "patch"},
+		},
+	}
+
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{Kind: "Role", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GoldmaneRoleName,
+			Namespace: GoldmaneNamespace,
+		},
+		Rules: policyRules,
+	}
+}
+
+func (c *Component) deploymentSelector() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app.kubernetes.io/name": GoldmaneDeploymentName,
+		},
+	}
+}
+
+func (c *Component) networkPolicy() *netv1.NetworkPolicy {
+	return &netv1.NetworkPolicy{
+		TypeMeta:   metav1.TypeMeta{Kind: "NetworkPolicy", APIVersion: "networking.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: "allow-goldmane", Namespace: GoldmaneNamespace},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: *c.deploymentSelector(),
+			PolicyTypes: []netv1.PolicyType{netv1.PolicyTypeIngress, netv1.PolicyTypeEgress},
+			Ingress: []netv1.NetworkPolicyIngressRule{
+				{
+					Ports: []netv1.NetworkPolicyPort{{
+						Protocol: ptr.ToPtr(corev1.ProtocolTCP),
+						Port:     ptr.ToPtr(intstr.FromInt32(GoldmaneServicePort)),
+					}},
+				},
+			},
+			Egress: []netv1.NetworkPolicyEgressRule{
+				{
+					To: []netv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
```

Comment:
Do we want a common LabelSelector function for this? Looks awfully similar to `deploymentSelector()`

Not required for this PR, just a thought.

Reply:
Done.